crates/matrix-sdk-base/src/client.rs

```diff
@@ -831,19 +790,32 @@ impl BaseClient {
                 room_info.mark_members_missing();
             }
 
+            let (raw_state_events_from_timeline, state_events_from_timeline) =
+                processors::state_events::collect_sync_from_timeline(
+                    &mut context,
+                    &new_info.timeline.events,
+                );
+
+            let mut other_new_user_ids = processors::state_events::dispatch_and_get_new_users(
+                &mut context,
+                (&raw_state_events_from_timeline, &state_events_from_timeline),
+                &mut room_info,
+                &mut ambiguity_cache,
+            )
+            .await?;
```

Comment:
Because `handle_timeline` is no longer handling the state events, we have to do this prior the call to `handle_timeline`, so here.